docs/src/quickstart.md

```diff
@@ -0,0 +1,93 @@
+# Quickstart
+
+## Writing your first kernel
+
+Kernel functions are marked with the [`@kernel`](@ref). Inside the `@kernel` macro
+you can use the [kernel language](@ref api_kernel_language). As an example, the `mul2`` kernel
+below will multiply each element of the array `A` by `2`. It uses the [`@index`](@ref) macro
+to obtain the global linear index of the current work item.
+
+```julia
+@kernel function mul2_kernel(A)
+  I = @index(Global)
+  A[I] = 2 * A[I]
+end
+```
+
+## Launching kernel on the host
+
+You can construct a kernel for a specific backend by calling the kernel
+function, with the first argument being the device of type `KA.Device`, the
+second argument being the size
+of the workgroup and the third argument being a static `ndrange`. The second and
+third arguments are optional. After instantiating the kernel, you can launch it by
+calling the kernel object with these arguments and some keyword arguments
+configuring the specific launch.
+
+```julia
+A = ones(1024, 1024)
+ev = mul2_kernel(CPU(), 16)(A, ndrange=size(A))
+wait(ev)
+all(A .== 2.0)
+```
+`mul_kernel(CPU(), 16)` creates a kernel for the host device `CPU()` with a static
+workgroup size of `16` and a dynamic `ndrange`. This function returns a kernel launched with the inputs
+and the dynamic `ndrange` as a keyword argument via `kernel(A,ndrange=size(A))`.
+The kernel
+eventually returning an event `ev`. All kernels are launched asynchronously.
+Thus, event `ev` specifies the current state of the execution.
+The [`wait`] blocks the *host* until the event `ev` has completed on the device. This implies that the host will launch no new kernels on any device until the wait returns.
+## Launching kernel on the device
+
+To launch the kernel on a CUDA supported GPU we generate the kernel
+for the device of type `CUDADevice` provided by `CUDAKernels` (see [backends](backends.md)).
+
+```julia
+using CUDAKernels # Required to access CUDADevice
+A = CUDA.ones(1024, 1024)
+ev = mul2_kernel(CUDADevice(), 16)(A, ndrange=size(A))
+wait(ev)
+all(A .== 2.0)
+```
```

Comment:
Also give an example for ROCM? There are configurations available in the example directory.

Reply:
Ok, I tried to put it in.